modules/osci/Makefile

```diff
@@ -46,146 +46,107 @@ OSCI_GIT_USER_NAME ?= "ACM CICD"
-	@mv tmp $(OSCI_MANIFEST_DIR)/$(OSCI_MANIFEST_FILENAME)
-
-.PHONY: osci/replace
-## Remove component named $OSCI_REPLACED_COMPONENT and add component named $OSCI_COMPONENT_NAME at version $OSCI_COMPONENT_VERSION in stage $OSCI_PIPELINE_STAGE
-osci/replace: %replace: %_clone %_replace
-	@$(SELF) osci/_push OSCI_GIT_MESSAGE="Replaced $(OSCI_REPLACED_COMPONENT) with $(OSCI_COMPONENT_NAME)"
-
-.PHONY: osci/_replace
-# Remove component named $OSCI_REPLACED_COMPONENT and add component named $OSCI_COMPONENT_NAME at version $OSCI_COMPONENT_VERSION in stage $OSCI_PIPELINE_STAGE
-osci/_replace: %_replace:
-	$(call assert-set,OSCI_COMPONENT_NAME)
-	$(call assert-set,OSCI_COMPONENT_VERSION)
-	$(call assert-set,OSCI_REPLACED_COMPONENT)
-	@$(SELF) osci/_delete OSCI_DELETED_COMPONENT=$(OSCI_REPLACED_COMPONENT)
-	@$(SELF) osci/_add
-
-.PHONY: osci/delete
-## Remove component named $OSCI_DELETED_COMPONENT from $OSCI_PIPELINE_STAGE's manifest pipeline
-osci/delete: %delete: %_clone %_delete
-	@$(SELF) osci/_push OSCI_GIT_MESSAGE="Deleted $(OSCI_DELETED_COMPONENT)"
-
-.PHONY: osci/_delete
-# Remove component named $OSCI_DELETED_COMPONENT from $OSCI_PIPELINE_STAGE's manifest pipeline
-osci/_delete: %_delete:
-	$(call assert-set,OSCI_DELETED_COMPONENT)
-	@$(OSCI_JQ) '[$(OSCI_DELETION_QUERY)]' $(OSCI_MANIFEST_DIR)/$(OSCI_MANIFEST_FILENAME) > tmp
-	@mv tmp $(OSCI_MANIFEST_DIR)/$(OSCI_MANIFEST_FILENAME)
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
 
 .PHONY: osci/publish
 ## Add or update component $OSCI_COMPONENT_NAME to have version $OSCI_COMPONENT_VERSION in the pipeline manifest in stage $OSCI_PIPELINE_STAGE
-osci/publish: %publish: %_clone %_publish
-	@$(SELF) osci/_push OSCI_GIT_MESSAGE="Updated $(OSCI_COMPONENT_NAME)"
-	@echo "Successfully updated ${OSCI_COMPONENT_NAME} to ${OSCI_COMPONENT_NAME}:${OSCI_COMPONENT_VERSION} in https://$(OSCI_PIPELINE_SITE)/$(OSCI_PIPELINE_ORG)/$(OSCI_PIPELINE_REPO)#$(OSCI_RELEASE_VERSION)-$(OSCI_PIPELINE_STAGE)"
-
-.PHONY: osci/_publish
-# Add or update component $OSCI_COMPONENT_NAME to have version $OSCI_COMPONENT_VERSION in the pipeline manifest in stage $OSCI_PIPELINE_STAGE
-osci/_publish: %_publish:
-	$(call assert-set,OSCI_COMPONENT_NAME)
-	$(call assert-set,OSCI_COMPONENT_VERSION)
-	@if [[ -z `$(SELF) -s osci/_read_alias` ]]; \
-	then echo "Component $(OSCI_COMPONENT_NAME) does not have an entry in $(OSCI_MANIFEST_DIR)/$(OSCI_IMAGE_ALIAS_FILENAME)" ; echo "Failing the build." ; exit 1; \
-	else echo "Component $(OSCI_COMPONENT_NAME) has an entry in $(OSCI_MANIFEST_DIR)/$(OSCI_IMAGE_ALIAS_FILENAME)"; \
-	fi
-	@echo `date`: Waiting $(OSCI_PUBLISH_DELAY) minutes for post commit images job to finish
-	@sleep $$(( $(OSCI_PUBLISH_DELAY) * 60 ))
-	@echo `date`: Done waiting
-	@cd $(OSCI_MANIFEST_DIR); $(GIT) pull --quiet
-	@if [ -z "$(SELF) -s osci/_read" ]; \
-	then $(SELF) osci/_add; \
-	else $(SELF) osci/_replace OSCI_REPLACED_COMPONENT=$(OSCI_COMPONENT_NAME); \
+osci/publish:
+	@echo $(OSCI_DATETIME) > DATETIME
```

Comment:
This is unconditional, where the old code would check for the existence of a `DATETIME` file before writing it.  In The Beginning, we used that to ensure a snapshot timestamp remained consistent throughout processing - I'm unsure if osci has any more moving parts during a snapshot, but if so, this should be surrounded with a check for `@if [ -f DATETIME ]`.

Reply:
There's nothing else going on. When a prow job is run, it starts in a new pod.